### PR TITLE
Issue/1721

### PR DESCRIPTION
**Pull request by mehul0810**
## Description
This PR is for #1721 

## How Has This Been Tested?
I have tested this with different scenario's:
1. Connecting and Disconnecting User from Donor Profile.
2. Disconnecting, Changing and Connecting User from Donor Profile.
3. Disconnecting User from Donor Profile and then try logging in. Then, user shouldnot be able to login and an email is sent to admin for reconnecting user.
4. Clicking on `Reconnect User` button in email should reconnect donor with user and display a success notice.

## Screenshots (jpeg or gifs if applicable):
image

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/process-donation.php [hunk 1]
```diff
@@ -715,6 +715,23 @@ function give_donation_form_validate_user_login() {
...
+
+					$disconnected_donor_id = get_user_meta( $user_data->ID, '_give_disconnected_donor_id', true );
+
+					// Send Email.
+					give_admin_email_user_donor_disconnection( $user_data->ID, $disconnected_donor_id );
+
+					// Donor and User are disconnected.
+					give_set_error(
+						'donor_user_disconnected',
+						__( 'Your Donor and User profile are no longer connected. Please contact site administrator.', 'give' )
```
Comment (DevinWalker) [inline includes/process-donation.php]: Change the text to:
> Your Donor and User profile are no longer connected. Please contact the site administrator.

**Comment by DevinWalker** [comment]
@mehul0810 let's discuss this when you come online. I have a few questions:

1. When a user and donor profile are disconnected this PR is preventing the user from logging into WordPress admin through the give login shortcode. However, if they simply access `/wp-admin` or `/wp-login.php` they will be able to login. Right? 
2. It's my thought that would shouldn't prevent them from logging in but rather prevent them from accessing the donations without proving their identity via email confirmation. So if I attempt to check my donations profile when logged in to a disconnected account I'll see something like "Your Donor and User profile are no longer connected. Please enter your email below to confirm your identity." The user then would receive an email at the _donor_ email address and then once clicked be able to access it. Does that make sense?

**Comment by mehul0810** [comment]
@DevinWalker Please check my comments

1. When a user and donor profile are disconnected this PR is preventing the user from logging into WordPress admin through the give login shortcode. However, if they simply access /wp-admin or /wp-login.php they will be able to login. Right?
-- **Yes. If user and donor profile are disconnected. Then, user can login via `wp-admin`. But, not via login in donation form.**

2. It's my thought that would shouldn't prevent them from logging in but rather prevent them from accessing the donations without proving their identity via email confirmation. So if I attempt to check my donations profile when logged in to a disconnected account I'll see something like "Your Donor and User profile are no longer connected. Please enter your email below to confirm your identity." The user then would receive an email at the donor email address and then once clicked be able to access it. Does that make sense?
-- **Yes. we can allow login user and restrict the Give Core area for donors like accessing donations, Give profile editor and other relating shortcodes.**

Need more info on why we want to send email to donor email address? I think donor and user email address will be same in most of the cases. We should send email to admin and admin should have those rights to allow donor access to user. Let me know your thoughts

**Comment by ravinderk** [comment]
@mehul0810  
While saving name with profile editor, I am getting error notice.
Let me know if you also able to produce it.

oyagxc0amr

**Comment by mehul0810** [comment]
@DevinWalker @ravinderk I have made changes to this PR.
1. Login via WP Admin and Shortcode no longer restrict user from logging in.
2. Processing Donation is restricted with message if user is logged in and disconnected with donor and tries to donate.
3. It will show a message and hide `give_profile_editor` form, if donor and user profile are disconnected.
4. For `donation_history` shortcode, there is a message already displayed when user and donor profile are disconnected. So, i have not added message to show as per the PR as it is not required in this case.

Please let me know your thoughts on the above points.

**Comment by mehul0810** [comment]
@DevinWalker I had a discussion @ravinderk yesterday and we decided to keep a question for discussion. 

What will happen if user and donor with email `hello@give.dev` is doing a guest checkout and donor-user profiles are disconnected?

Let me know your thoughts on this.

**Comment by DevinWalker** [comment]
@mehul0810 my thought is that the donation goes through. We don't want to prevent a donation from happening.

**Comment by mehul0810** [comment]
@DevinWalker I have removed the restriction while processing donation. So, now, donation process will go through irrespective of user being logged in or non logged in, when donor and user profile are disconnected.

**Pull request merged**